controllers/cluster_controller.go

```diff
@@ -357,6 +383,32 @@ func (r *ClusterReconciler) postClusterProviderReconcile(ctx context.Context, lo
 	return controller.Result{}, nil
 }
 
+func (r *ClusterReconciler) updateStatus(ctx context.Context, log logr.Logger, cluster *anywherev1.Cluster) error {
+	// When EKS-A cluster is fully deleted, we do not need to update the status. Without this check
+	// the subsequent patch operations would fail if the status is updated after it is fully deleted.
+	if !cluster.DeletionTimestamp.IsZero() && len(cluster.GetFinalizers()) == 0 {
+		log.Info("Cluster is fully deleted, skipping cluster status update")
+		return nil
+	}
+
+	log.Info("Updating cluster status")
+
+	if err := clusters.UpdateControlPlaneStatus(ctx, r.client, cluster); err != nil {
```

Comment:
this name is a bit confusing
We are not updating the control plane status
we are updating the cluster status part that reflects the state of the cluster control plane
right?

Reply:
yes, i've updated the name to reflect that purpose better.